### PR TITLE
Docs: Link tokens in the format string grammars

### PR DESCRIPTION
**Pull request by wjandrea**
Fix links from Format String Syntax grammar and Format Specification Mini-Language grammar to Python grammar.
Also link `arg_name` token within Format String Syntax grammar.

---

Please check my work. I'm not very familiar with RST, just read [the docs on `productionlist`s](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-productionlist) and did my best.

Should be backported to all current versions.

----
:books: Documentation preview :books:: https://cpython-previews--108184.org.readthedocs.build/en/108184/library/string.html#format-string-syntax

### CODE REVIEW

**Comment by wjandrea** [comment]
Looks like the links to `python-grammar` didn't work. I'm not sure if that's cause the test failed or what.

**Comment by AA-Turner** [comment]
See #109209 and #109210.

**Review by AA-Turner — APPROVED** [review]

**Comment by skirpichev** [comment]
@wjandrea, could you, please, resolve merge conflicts?

BTW, shouldn't we also bug sphinx-doc (I don't see related issue(s))?

**Comment by skirpichev** [comment]
FYI: sphinx-doc/sphinx#11854

**Comment by wjandrea** [comment]
@skirpichev resolved

File: Doc/library/string.rst [hunk 1]
```diff
@@ -208,10 +208,10 @@ The grammar for a replacement field is as follows:
...
-   arg_name: [`identifier` | `digit`+]
-   attribute_name: `identifier`
-   element_index: `digit`+ | `index_string`
+   field_name: `arg_name` ("." `attribute_name` | "[" `element_index` "]")*
+   arg_name: [`~python-grammar:identifier` | `~python-grammar:digit`+]
+   attribute_name: `~python-grammar:identifier`
+   element_index: `~python-grammar:digit`+ | `index_string`
    index_string: <any source character except "]"> +
    conversion: "r" | "s" | "a"
    format_spec: <described in the next section>
```
Comment (skirpichev) [inline Doc/library/string.rst]:
```suggestion
   format_spec: `format-spec:format_spec`
```
?

File: Doc/tools/extensions/pyspecific.py [hunk 2]
```diff
@@ -48,6 +48,10 @@
     Body.enum.converters['lowerroman'] = \
     Body.enum.converters['upperroman'] = lambda x: None
 
+# monkey-patch the productionlist directive to allow hyphens in group names
```
Comment (skirpichev) [inline Doc/tools/extensions/pyspecific.py:51]: Maybe you could reference the sphinx issue here.

**Review by skirpichev — APPROVED** [review]

**Comment by skirpichev** [comment]
LGTM, except for two nitpicks

@AA-Turner?

File: Doc/tools/extensions/pyspecific.py (hunk 2 above)
Comment (AA-Turner) [inline Doc/tools/extensions/pyspecific.py:51]:
```suggestion
# monkey-patch the productionlist directive to allow hyphens in group names
# https://github.com/sphinx-doc/sphinx/issues/11854
```

**Pull request merged**